src/components/app/DocumentList/index.js

```diff
@@ -0,0 +1,855 @@
+            parentWindowId,
+            headerProperties,
+          });
+        }
+
+        indicatorState('saved');
+      })
+      .catch(() => {
+        this.setState({ triggerSpinner: false });
+      });
+  };
+
+  getLocationData = (resultById) => {
+    const {
+      windowType,
+      viewId,
+      reduxData: { mapConfig },
+    } = this.props;
+
+    locationSearchRequest({ windowId: windowType, viewId }).then(({ data }) => {
+      const locationData = data.locations.map((location) => {
+        const name = get(
+          resultById,
+          [location.rowId, 'C_BPartner_ID', 'value', 'caption'],
+          location.rowId
+        );
+
+        return {
+          ...location,
+          name,
+        };
+      });
+
+      if (locationData.length) {
+        updateViewData(locationData, windowType);
+      }
+
+      if (mapConfig && mapConfig.provider) {
+        // for mobile show map
+        // for desktop show half-n-half
+        this.setState({ panelsState: GEO_PANEL_STATES[1] });
+      }
+    });
+  };
+
+  // MANAGING SORT, PAGINATION, FILTERS --------------------------------------
+
+  /**
+   * @method handleChangePage
+   * @summary ToDo: Describe the method.
+   */
+  handleChangePage = (index) => {
+    const { reduxData } = this.props;
+    let currentPage = reduxData.page;
+
+    switch (index) {
+      case 'up':
+        currentPage * reduxData.pageLength < reduxData.size
+          ? currentPage++
+          : null;
+        break;
+      case 'down':
+        currentPage != 1 ? currentPage-- : null;
+        break;
+      default:
+        currentPage = index;
+    }
+
+    this.setState(
+      {
+        triggerSpinner: true,
+      },
+      () => {
+        this.getData(reduxData.viewId, currentPage, reduxData.sort);
+      }
+    );
+  };
+
+  /**
+   * @method sortData
+   * @summary ToDo: Describe the method.
+   */
+  sortData = (asc, field, startPage) => {
+    const { viewId, page } = this.props;
+
+    this.setState(
+      {
+        triggerSpinner: true,
+      },
+      () => {
+        this.getData(viewId, startPage ? 1 : page, getSortingQuery(asc, field));
+      }
+    );
+  };
+
+  /**
+   * @method handleFilterChange
+   * @summary ToDo: Describe the method.
+   */
+  handleFilterChange = (activeFilters) => {
+    const locationSearchFilter = activeFilters.has(`location-area-search`);
+
+    // TODO: filters should be kept in the redux state
+    this.setState(
+      {
+        filtersActive: activeFilters,
+        triggerSpinner: true,
+      },
+      () => {
+        this.fetchLayoutAndData(true, locationSearchFilter);
+      }
+    );
+  };
+
+  /**
+   * @method resetInitialFilters
+   * @summary ToDo: Describe the method.
+   */
+  resetInitialFilters = (filterId, parameterName) => {
+    let { initialValuesNulled } = this.state;
+    let filterParams = initialValuesNulled.get(filterId);
+
+    if (!filterParams && parameterName) {
+      filterParams = Set([parameterName]);
```

Comment:
maybe we should use same pattern and for Set to use iSet ? wdyt ?

Reply:
`iSet` ? What's that ?

Reply:
iMap I would like to be the immutable map and iSet the immutable set. Can you please update these within the file . Like i said above it's easy to spot these in code. Seeing iMap you'l know that is no ES Map and is the immutable one. Makes sense ?